actions/watch.ts

```diff
@@ -83,101 +97,123 @@ export const checkForAndPlaceOrder: ActionFn = async (
   const blockEvent = event as BlockEvent;
   const registry = await Registry.load(context, blockEvent.network);
   const chainContext = await ChainContext.create(context, blockEvent.network);
-
-  console.log(`Processing block ${blockEvent.blockNumber}...`);
+  const { network } = blockEvent;
 
   // enumerate all the owners
   for (const [owner, conditionalOrders] of registry.ownerOrders.entries()) {
-    console.log(`Checking ${owner}...`);
-
+    const deletedOrders = [];
     // enumerate all the `ConditionalOrder`s for a given owner
     for (const conditionalOrder of conditionalOrders) {
-      console.log(`Checking params ${conditionalOrder.params}...`);
+      // console.log(`Checking params ${conditionalOrder.params}...`);
       const contract = ComposableCoW__factory.connect(
         conditionalOrder.composableCow,
         chainContext.provider
       );
-      try {
-        const { order, signature } =
-          await contract.callStatic.getTradeableOrderWithSignature(
-            owner,
-            conditionalOrder.params,
-            "0x",
-            conditionalOrder.proof ? conditionalOrder.proof.path : []
-          );
 
-        const orderToSubmit: Order = {
-          ...order,
-          kind: kindToString(order.kind),
-          sellTokenBalance: balanceToString(order.sellTokenBalance),
-          buyTokenBalance: balanceToString(order.buyTokenBalance),
-        };
+      const { deleteConditionalOrder } = await _checkForAndPlaceOrder(
+        owner,
+        network,
+        conditionalOrder,
+        contract,
+        chainContext
+      );
 
-        // calculate the orderUid
-        const orderUid = computeOrderUid(
-          {
-            name: "Gnosis Protocol",
-            version: "v2",
-            chainId: blockEvent.network,
-            verifyingContract: GPV2SETTLEMENT,
-          },
-          {
-            ...orderToSubmit,
-            receiver:
-              orderToSubmit.receiver === ethers.constants.AddressZero
-                ? undefined
-                : orderToSubmit.receiver,
-          },
-          owner
-        );
+      if (deleteConditionalOrder) {
+        deletedOrders.push(conditionalOrder);
+      }
+    }
 
-        // if the orderUid has not been submitted, or filled, then place the order
-        if (!conditionalOrder.orders.has(orderUid)) {
-          console.log(
-            `Placing orderuid ${orderUid} with Order: ${JSON.stringify(order)}`
-          );
+    deletedOrders.forEach((conditionalOrder) =>
+      conditionalOrders.delete(conditionalOrder)
+    );
+  }
 
-          await placeOrder(
-            { ...orderToSubmit, from: owner, signature },
-            chainContext.api_url
-          );
+  // Update the registry
+  await registry.write();
+};
+
+async function _checkForAndPlaceOrder(
```

Comment:
This was not necesary, but I felt the main loop was doing way too many things, so I broke down the logic a bit. It was hard for me to reason about some errors I was getting (see PR description)